packages/cli/src/executions/executions.controller.ts

```diff
@@ -110,4 +110,23 @@ export class ExecutionsController {
 
 		return await this.executionService.delete(req, workflowIds);
 	}
+
+	@Patch('/:id')
+	async update(req: ExecutionRequest.Update) {
+		if (!isPositiveInteger(req.params.id)) {
+			throw new BadRequestError('Execution ID is not a number');
+		}
+
+		const workflowIds = await this.getAccessibleWorkflowIds(req.user, 'workflow:read');
+
+		if (workflowIds.length === 0) throw new NotFoundError('Execution not found');
```

Comment:
Shouldn't we instead check that the user has access to the given workflow for which the execution belongs to?

Reply:
As far as I understood this particular check is more like a fail-fast mechanism, to stop handling the request if user has no access to any workflows. I've added the actual access check as well.